CRM/Civiquickbooks/Form/Settings.php

```diff
@@ -139,11 +139,25 @@ public function getFormSettings() {
   public function saveSettings() {
     $settings = $this->getFormSettings();
     $values = array_intersect_key($this->_submittedValues, $settings);
-    $previousValues = CRM_Quickbooks_APIHelper::getQuickBooksCredentials();
+
+    // Fix for unsetting a checkbox. When setting a checkbox,
+    // quickbooks_bool_value => 1 is returned. But when unsetting a checkbox,
+    // quickbooks_bool_value => 0 is NOT returned. Absence of a checkbox
+    // attribute should be interpreted as setting its value to 0.
+    foreach ($settings as $setting_name => $setting_info) {
+        if ($setting_info['type'] == "Boolean" && !array_key_exists($setting_name, $values)) {
+            $values[$setting_name] = 0;
+        }
+    }
 
     civicrm_api3('setting', 'create', $values);
 
-    if ($previousValues['clientID'] != 'quickbooks_consumer_key' || $previousValues['clientSecret'] != 'quickbooks_shared_secret') {
```

Comment:
Need to confirm the issue & do run testing on this section before we can merge.

Reply:
ok sounds like I should leave this one out for now? I have a bunch of other more relevant code I'd love to PR instead